libs/common/src/main/java/org/opensearch/common/annotation/DeprecatedApi.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+/**
+ * Marks the public APIs as deprecated and scheduled for removal in one of the upcoming
+ * major releases. The types marked with this annotations could only be other {@link PublicApi}s.
+ *
+ * @opensearch.api
+ */
+@Documented
+@Target({ ElementType.TYPE, ElementType.PACKAGE, ElementType.METHOD, ElementType.CONSTRUCTOR })
+public @interface DeprecatedApi {
```

Comment:
Add '@PublicApi` annotation?

Reply:
That would not be necessary, example: the `@PublicApi` is applied on class level whereas `@DeprecatedApi` could be applied to individual method within API. Or I misunderstood you comment - apply `@PublicApi` to all annotations here?

Reply:
Getting nitpicky, this is a chase the tail game :D

Reply:
Tail is chased :D